addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/handler/FoobotHandler.java

```diff
@@ -0,0 +1,340 @@
+import org.openhab.binding.foobot.internal.FoobotConfiguration;
+import org.openhab.binding.foobot.internal.json.FoobotJsonData;
+import org.openhab.binding.foobot.internal.json.FoobotJsonResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link FoobotHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Divya Chauhan - Initial contribution
+ */
+public class FoobotHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FoobotHandler.class);
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private static final String URL1 = "https://api.foobot.io/v2/owner/%username%/device/";
+
+    private static final String URL2 = "https://api.foobot.io/v2/device/%uuid%/datapoint/%period%/last/0/?sensorList=%sensors%";
+
+    FoobotJsonData foobotData;
+
+    private Gson gson;
+
+    private String uuid;
+
+    FoobotConfiguration config;
+
+    private ScheduledFuture<?> refreshJob;
+
+    public FoobotHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Foobot handler.");
+        config = getConfigAs(FoobotConfiguration.class);
+
+        logger.debug("config apikey = {}", config.apikey);
+        logger.debug("config apikey = {}", config.username);
+        logger.debug("config apikey = {}", config.mac);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.username) == null) {
+            errorMsg = "Parameter 'username' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.mac) == null) {
+            errorMsg = "Parameter 'Mac Address' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh != null && config.refresh < 5) {
+            errorMsg = "Parameter 'refresh' must be at least 5 minutes";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+
+            try {
+                String urlStr = URL1.replace("%username%", StringUtils.trimToEmpty(config.username));
+                logger.debug("URL = {}", urlStr);
+
+                // Run the HTTP request and get the list of all foobot devices belonging to the user from foobot.io
+
+                URL url = new URL(urlStr);
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Please don't use UrlConnection for http communication as they are subject to resource leaks and hard to get right. For such simple calls you may use `HttpUtil` for your convenience - or [the Jetty client](https://www.eclipse.org/smarthome/documentation/development/bindings/dependencies.html#for-http-operations) for more complicated stuff.